main.py

```diff
@@ -27,86 +32,147 @@
     password=None
 )
 
+class PrintJob(Enum):
+    COMPLETED = 0
+    PENDING = 1
+    FILE_ERROR = 2
+    QUOTA_LIMIT = 3
+    JOB_ERROR = 4
+
+print_jobs = {name: {} for name in CONFIG.PRINTERS.NAMES}
+
+class Job():
+    def cleanup():
+        global print_jobs
+        def notdone(job):
+            status = job.current_status()
+            if status == PrintJob.COMPLETED:
+                return job.last_updated + CONFIG.PERSIST_TIME.COMPLETED > time.time()
+            elif status == PrintJob.PENDING:
+                return job.last_updated + CONFIG.PERSIST_TIME.DEFAULT > time.time()
+            return job.last_updated + CONFIG.PERSIST_TIME.ERROR > time.time()
+        
+        print_jobs = {
+            printer: {
+                id: job for id, job in jobs.items() if notdone(job)
+            } for printer, jobs in print_jobs.items()
+        }
+
+    def get_recent(time):
+        jobs = {}
+        for printer_name, printer in print_jobs.items():
+            jobs[printer_name] = [job.toJSON() for job_id, job in printer.items() if job.last_updated > time]
+        return jobs
+
+    def add(printer_name, username, time, status, job_id):
+        if job_id in print_jobs[printer_name]:
+            print_jobs[printer_name][job_id].update(username, status, time)
```

Comment:
Is there a difference between updating a job and just creating a new Job object assuming you check for username collision? This ties into a more general statement about whether we care about previous statuses at all, seeing as we only ever return the latest status.

Reply:
Previous statuses are good to retain the time flow of status codes.